storage/backend/ceph/internals.go

```diff
@@ -6,14 +6,46 @@ import (
 	"regexp"
 	"strings"
 
+	"encoding/json"
+
 	"github.com/contiv/volplugin/storage"
 
 	log "github.com/Sirupsen/logrus"
 )
 
 func (c *Driver) mapImage(do storage.DriverOptions) (string, error) {
+	var rbdMaps map[string] struct {
+		Pool string `json:"pool"`
+		Name string `json:"name"`
+		Device string `json:"device"`
+	}
+
+	var device string
+
 	blkdev, err := exec.Command("rbd", "map", do.Volume.Name, "--pool", do.Volume.Params["pool"]).Output()
-	device := strings.TrimSpace(string(blkdev))
+	device = strings.TrimSpace(string(blkdev))
+
+	if device == "" && err == nil {
+		output, err := exec.Command("rbd", "showmapped", "--format", "json").Output()
+
+		if err == nil {
```

Comment:
this is just a minor code styling nit, by doing a `err != nil` check earlier, the large part of the code within the `if` can be kept indented to left, which is also easier t read.

```
if err != nil {
    return "", err
}

... the main logic left indented goes here
...
```